src/components/Table/GridTable.tsx

```diff
@@ -274,6 +290,111 @@ export function GridTable<R extends Kinded, X extends Only<GridTableXss, X> = an
   const expandedColumnIds: string[] = useComputed(() => tableState.expandedColumnIds, [tableState]);
   const columnSizes = useSetupColumnSizes(style, columns, resizeTarget ?? resizeRef, expandedColumnIds);
 
+  function isCursorBelowMidpoint(target: HTMLElement, clientY: number) {
+    const style = window.getComputedStyle(target);
+    const rect = target.getBoundingClientRect();
+
+    const pt = parseInt(style.getPropertyValue("padding-top")) / 2;
+    const pb = parseInt(style.getPropertyValue("padding-bottom"));
+
+    return clientY > rect.top + pt + (rect.height - pb) / 2;
+  }
+
+  // allows us to unset children and grandchildren, etc.
+  function recursiveSetDraggedOver(rows: GridDataRow<R>[], draggedOver: DraggedOver) {
+    rows.forEach((r) => {
+      tableState.maybeSetRowDraggedOver(r.id, draggedOver);
+      if (r.children) {
+        recursiveSetDraggedOver(r.children, draggedOver);
+      }
+    });
+  }
+
+  function onDragStart(row: GridDataRow<R>, evt: DragEventType) {
```

Comment:
Suggestion: I think `useCallback`-ing all of these functions will be necessary to keep the all rows from re-rendering on each GridTable render. As-is the `Row` props have different values/new functions on each GridTable render.

...it might be "too cute" but fwiw that `RowStates` / `TableState` classes are effectively already cached/1:1 with each GridTable, so if you did something like `TableState.onDragStart` (all of these become methods on `TableState` or `RowStates`), and then inside of of `Row` you could do `onDragStart => tableState.onDragStart(row, evt)`.

I guess you'd need the `setDraggedRow` setter, but you could do something like a `TableState.draggedRow` field, that just gets set... (i.e. skip the `useState` + `useRef` pair, and just use a field on the mobx `TableState` to achieve basically the same thing...

Reply:
I added a test to GridTable.test.js to look at row rerenders for draggable rows and I believe they already only rerender on data change, just like the non-draggable rows, with the exception that they need to rerender when I'm setting css for the spacer and when they get reordered.